src/traits/padding.rs

```diff
@@ -32,7 +32,7 @@ pub trait PaddingScheme {
 /// Digital signature scheme.
 pub trait SignatureScheme {
     /// Sign the given digest.
-    fn sign<Rng: CryptoRngCore>(
+    fn sign<Rng: CryptoRng>(
```

Comment:
I think you should use `TryCryptoRng` in all functions/methods which return `Result`.

Reply:
But I can't do that because the rng is passed down to `crypto-bigint` which expects a `CryptoRng`. I can `UnwrapMut` on the rng, but that will just throw runtime panics.

I think I'd rather have the consumer do the `UnwrapMut` themselves.

Reply:
specifically, we always end up in:
https://github.com/RustCrypto/RSA/blob/9e7faa42bf84ee5a8487012303cf50f297f91f4c/src/algorithms/rsa.rs#L34

Which then calls:
https://github.com/RustCrypto/RSA/blob/9e7faa42bf84ee5a8487012303cf50f297f91f4c/src/algorithms/rsa.rs#L145

which calls into crypto-bigint:
https://github.com/RustCrypto/RSA/blob/9e7faa42bf84ee5a8487012303cf50f297f91f4c/src/algorithms/rsa.rs#L161

And I don't have a `try_random_mod` in crypto-bigint. It specifically wants an `RngCore`

Reply:
I don't see any obstacles to adding `try_random_mod`. It should be done either way.

Reply:
yeah, you're right, it's actually fairly easy.
https://github.com/RustCrypto/crypto-bigint/pull/770